llpc/builder/llpcBuilderImplImage.cpp

```diff
@@ -929,19 +939,24 @@ Value* BuilderImplImage::CreateImageLoad(
                             derivatives);
 
     // The intrinsics insist on an FP data type, so we need to bitcast if we want an integer data type.
-    Type* pIntrinsicDataTy = pResultTy;
     if (pTexelTy->isIntOrIntVectorTy())
     {
-        pIntrinsicDataTy = ConvertToFloatingPointType(pTexelTy);
-        if (pTexelTy != pResultTy)
-        {
-            // TFE
-            pIntrinsicDataTy = StructType::get(pTexelTy->getContext(), { pTexelTy, getInt32Ty() });
-        }
+        pTexelTy = ConvertToFloatingPointType(pTexelTy);
+    }
```

Comment:
Can't we remove this entirely? amdgcn.image.load intrinsics support both int and float -- if the LLPC "intrinsics" don't, could you please help to fix that?

Reply:
Sure, I will explore this and try to avoid bitcast from integer to float.